### ISSUE TITLE
[Examples] Latest PyTorch 2.2.0 fail to work on default VM image

### ISSUE THREAD

**Issue by Michaelvll** [skypilot-org/skypilot#3070]
The latest PyTorch 2.2 fails to work on our default VM image, such as on GCP. We should either update version or update the examples with the older pytorch.

Both of ` pip install torch torchvision` and `pip install torch torchvision --index-url https://download.pytorch.org/whl/cu118` fail to work on a new cluster launched on GCP.

(from the [README](https://github.com/skypilot-org/skypilot?tab=readme-ov-file#skypilot-in-1-minute))

```yaml
resources:
  accelerators: V100:1  # 1x NVIDIA V100 GPU

num_nodes: 1  # Number of VMs to launch

# Working directory (optional) containing the project codebase.
# Its contents are synced to ~/sky_workdir/ on the cluster.
workdir: ./torch_examples

# Commands to be run before executing the job.
# Typical use: pip install -r requirements.txt, git clone, etc.
setup: |
  pip install torch torchvision --index-url https://download.pytorch.org/whl/cu118

# Commands to run as a job.
# Typical use: launch the main program.
run: |
  cd mnist
  python main.py --epochs 1

```

**Comment by Michaelvll** [comment]
Another side issue with this: the output of the job does not include the `Segment fault` error message, although it will show when `ssh` into the machine and run the scrip manually.

**Comment by cblmemo** [comment]
Fixed by #3868. Closing now
